### PR TITLE
Sort editor help documentation methods and properties by alphabetical order

### LINKED ISSUE #35347: Doc items breaking into groups by first sentence rather than alphabetically
**Godot version:**
3.1 - 3.2rc1

**Issue description:**
Looks like incorrect spacing where it shouldn't and not breaking where it should:
image

### PR DESCRIPTION
**Pull request by ChristopheLY**
Closes https://github.com/godotengine/godot/issues/35347.

### CODE REVIEW

**Comment by Calinou** [comment]
Can you take a before/after screenshot so we can see the difference? Reading the description, I'm not sure if grouping by first letter is actually an upgrade.

**Comment by asheraryam** [comment]
I like the change, it makes it easier to navigate visually quickly. It's definitely an ergonomic change if nothing else.

**Comment by Calinou** [comment]
Looks good to me as well.

File: editor/editor_help.cpp [hunk 1]
```diff
@@ -517,7 +517,26 @@ void EditorHelp::_update_doc() {
 		class_desc->push_table(2);
 		class_desc->set_table_column_expand(1, true);
 
+		String first_char;
 		for (int i = 0; i < cd.properties.size(); i++) {
+			const String new_char = cd.properties[i].name.substr(0, 1);
```
Comment (akien-mga) [inline editor/editor_help.cpp:522]: `substr(0, 1)` is a convoluted way to do `[0]` ;)

**Comment by akien-mga** [comment]
Commits should be squashed into one: https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase

**Comment by ChristopheLY** [comment]
Should I close de PR now?

**Comment by Calinou** [comment]
@ChristopheLY No, it will be closed automatically when it's merged.

**Comment by ChristopheLY** [comment]
may fixe, #35347

**Comment by aaronfranke** [comment]
@ChristopheLY Is this still desired? If so, it needs to be [rebased](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase) and tested on the latest master branch.

**Comment by YuriSizov** [comment]
> I'm not sure if grouping by first letter is actually an upgrade.

I'm not convinced either. The API is just too varied and grouping by the first letter is bound to introduce a lot of unnecessary splitting in some classes. We could probably do something smarter, and try to separate large groups of methods based on a prefix, while grouping smaller chunks together regardless of their name.

It would be nice to fix the linked issue though. But this PR needs to be reworked against the current master.

**Comment by AThousandShips** [comment]
This shouldn't be relevant any longer as this was solved in past PRs like:
* https://github.com/godotengine/godot/pull/77554

**Pull request closed without merging**